src/Marten/Linq/SoftDeletes/SoftDeletedExtensions.cs

```diff
@@ -8,10 +8,8 @@ public static class SoftDeletedExtensions
     ///     The search results should include all documents, whether
     ///     soft-deleted or not
     /// </summary>
-    /// <typeparam name="T"></typeparam>
-    /// <param name="doc"></param>
     /// <returns></returns>
-    public static bool MaybeDeleted(this object doc)
+    public static bool MaybeDeleted<T>(this T doc)
```

Comment:
Noting that this is a binary breaking change, so updates w/o re-compilation won't work.

Adding new overloads with the generic signature wouldn't be breaking change, and the C# compiler has to bind to the generic overload (as it's more specific than the `object` overload).